addons/web/static/src/views/fields/relational_utils.js

```diff
@@ -495,7 +495,7 @@ export function useOpenMany2XRecord({
         }
 
         const { create: canCreate, write: canWrite } = activeActions;
-        const mode = (resId ? canWrite : canCreate) ? "edit" : "readonly";
+        const readonly = !(resId ? canWrite : canCreate);
```

Comment:
weird to allow opening a FormViewDialog with no id if canCreate is false... Having such a view in readonly doesn't make sense